core/src/main/java/com/yahoo/oak/NativeMemoryAllocator.java

```diff
@@ -193,7 +193,7 @@ public void readByteBuffer(Slice s) {
         assert blockID > NativeMemoryAllocator.INVALID_BLOCK_ID :
                 String.format("Invalid block-id: %s", s);
         Block b = blocksArray[blockID];
-        s.setBuffer(b.getBuffer());
+        s.setBuffer(b.getmemAdress(), b.getCapacity());
```

Comment:
I would also search through the Oak code for the world "ByteBuffer" (without matching the letters case) and see if it is used correctly. Because after your change there should be no internal ByteBuffer usage in Oak...